posthog/temporal/workflows/s3_batch_export.py

```diff
@@ -96,8 +97,16 @@ async def insert_into_s3_activity(inputs: S3InsertInputs):
             aws_access_key_id=inputs.aws_access_key_id,
             aws_secret_access_key=inputs.aws_secret_access_key,
         )
-        multipart_response = s3_client.create_multipart_upload(Bucket=inputs.bucket_name, Key=key)
-        upload_id = multipart_response["UploadId"]
+        details = activity.info().heartbeat_details
+
+        if len(details) == 2:
+            interval_start, upload_id = details
```

Comment:
I guess we have to be careful here re changing schema i.e. if we decide we need to change these values we'll have issues?

Reply:
We may also need the part number

Reply:
Yeah, there is no schema enforcement, temporal just takes a `Sequence[Any]`.

Reply:
I do want to hide all of this in the class I'm working on to handle S3 uploads so that at least we only have to deal with the schema in one place.

Reply:
And true, we need the part number and all the ETAGs so far...